src/page-modules/assistant/server/journey-planner/index.ts

```diff
@@ -784,3 +814,21 @@ async function getSortedViaTrips(
   };
   return sortedData;
 }
+
+function createPublicCodeLineMap(lines: LineFragment[]) {
+  const publicCodeLineList = new Map<string, string[]>();
+  lines.forEach((line) => {
+    if (line.publicCode) {
+      if (publicCodeLineList.has(line.publicCode)) {
+        publicCodeLineList.get(line.publicCode)?.push(line.id);
+      } else {
+        publicCodeLineList.set(line.publicCode, [line.id]);
+      }
+    }
+  });
+
+  // @ts-ignore
+  // @ts-expect-error: This is needed to allow iteration through Map objects
+  // @ts-nocheck
+  return JSON.stringify([...publicCodeLineList]);
```

Comment:
Is it necessary to stringify this?

Reply:
Yes, I think it is necessary. As I understand it, Map objects are not directly serializable to the JSON format, as JSON only supports simple data types such as strings, numbers, booleans, arrays and objects. Therefore, the Map object must be converted to a serializable form such as a string before it is sent to the client. On the client side, the string is turned into the etmap object again.

Reply:
Does it need to be a map? Seems like this can just be a normal object without actually loosing any functionality.